crates/paths/src/paths.rs

```diff
@@ -78,7 +74,26 @@ pub fn config_dir() -> &'static PathBuf {
             }
             .join("zed")
         } else {
-            home_dir().join(".config").join("zed")
+            let xdg_config_home = std::env::var_os("XDG_CONFIG_HOME").and_then(|val| {
```

Comment:
Rather than check these directly, I'd rather we used `dirs::config_dir()`

Reply:
Sorry but `dirs::config_dir()` is already used as a fallback in L84 below. And the point here is to support  `XDG_CONFIG_HOME` on non-Linux platforms.

Reply:
Btw `dirs` crate does not support XDG on non-Linux platforms. And if you are not familiar with the use of XDG spec on non-Linux platforms, I had a discussion recently about it in https://github.com/prefix-dev/pixi/issues/3757 where some context may help.